src/main/java/seedu/duke/lesson/LessonList.java

```diff
@@ -35,4 +38,24 @@ public void clearLessonList() {
     public void deleteLesson(int index) {
         lessonList.remove(index);
     }
+
+    public Lesson getLesson(int index) throws DukeException {
+        try {
+            return lessonList.get(index);
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException(Message.ERROR_INDEX_OUT_OF_BOUNDS);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(Message.ERROR_NUMBER_FORMAT);
+        }
+    }
+
+    public LessonList findLesson(String keyword) {
+        LessonList matchingLessonList = new LessonList();
+        for (Lesson lesson: lessonList) {

```

Comment:
Minor coding standard violation, more specifically the white space within a statement. So it should be written like this

```java
for (Lesson lesson : lessonList)
```